### PR TITLE
fix: map_unordered type signature errors

### LINKED ISSUE #35841: [Core]: map_unordered type signature errors
### What happened + What you expected to happen

vscode / pyright type error:

```
test.py:23:34 - error: Argument of type "(a: Unknown, v: Unknown) -> Unknown" cannot be assigned to parameter "fn" of type "() -> Any" in function "map_unordered"
```

### Versions / Dependencies

ray 2.4.0

### Reproduction script

```python
import ray
from ray.util import ActorPool

@ray.remote
class Predictor:
    def __init__(self):
        print("__init__")


    def predict(self, id: str) -> int:
        print("predict")
        return 1

def main():
    # create actors
    print("creating actors")
    predictors = [Predictor.remote() for _ in range(10)]
    pool = ActorPool(predictors)

    ids = [1,2,3]

    print("submit")
    results = pool.map_unordered(lambda a, v: a.predict.remote(v), ids) # <--- type error here

    print("wait on results")
    for r in results:
        print(r)

if __name__ == "__main__":
    main()
```

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by tekumara**
## Related issue number

Closes #35841
 
## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/util/actor_pool.py [hunk 1]
```diff
@@ -107,7 +110,9 @@ def get_generator():
 
         return get_generator()
 
-    def map_unordered(self, fn: Callable[[Any], Any], values: List[Any]):
+    def map_unordered(
```
Comment (jjyao) [inline python/ray/util/actor_pool.py:113]: Nice. Could you fix `def map()` as well?

**Review by jjyao — COMMENTED** [review]
Could you also rebase with master? Many test failures might have been fixed already.

File: python/ray/util/actor_pool.py (hunk 1 above)
Comment (tekumara) [reply python/ray/util/actor_pool.py:113]: 👍 good idea, now added.

File: python/ray/util/actor_pool.py [hunk 2]
```diff
@@ -1,7 +1,10 @@
-from typing import List, Callable, Any
+from typing import List, Callable, Any, TypeVar
 
 import ray
 from ray.util.annotations import DeveloperAPI
+import ray.actor
```
Comment (jjyao) [inline python/ray/util/actor_pool.py:5]: You don't need this import

**Review by jjyao — APPROVED** [review]

File: python/ray/util/actor_pool.py (hunk 2 above)
Comment (tekumara) [reply python/ray/util/actor_pool.py:5]: Without I get the following in vscode/pylance:

CleanShot 2023-06-17 at 10 11 54@2x

**Comment by jjyao** [comment]
@tekumara could you rebase with master again?

**Comment by tekumara** [comment]
thanks @jjyao i've now rebased (sorry for the notification spam that came along with this)

**Comment by jjyao** [comment]
@tekumara seems there is a circular import
 
```
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/anyscale-ray/envs/35843/lib/python3.9/site-packages/sphinx/config.py", line 329, in eval_config_file
    exec(code, namespace)
  File "/home/docs/checkouts/readthedocs.org/user_builds/anyscale-ray/checkouts/35843/doc/source/conf.py", line 31, in <module>
    import ray
  File "/home/docs/checkouts/readthedocs.org/user_builds/anyscale-ray/checkouts/35843/python/ray/__init__.py", line 117, in <module>
    from ray._private.state import (  # noqa: E402,F401
  File "/home/docs/checkouts/readthedocs.org/user_builds/anyscale-ray/checkouts/35843/python/ray/_private/state.py", line 15, in <module>
    from ray.util.annotations import DeveloperAPI
  File "/home/docs/checkouts/readthedocs.org/user_builds/anyscale-ray/checkouts/35843/python/ray/util/__init__.py", line 9, in <module>
    from ray.util.actor_pool import ActorPool
  File "/home/docs/checkouts/readthedocs.org/user_builds/anyscale-ray/checkouts/35843/python/ray/util/actor_pool.py", line 5, in <module>
    import ray.actor
  File "/home/docs/checkouts/readthedocs.org/user_builds/anyscale-ray/checkouts/35843/python/ray/actor.py", line 8, in <module>
    import ray._private.worker
  File "/home/docs/checkouts/readthedocs.org/user_builds/anyscale-ray/checkouts/35843/python/ray/_private/worker.py", line 50, in <module>
    import ray._private.serialization as serialization
  File "/home/docs/checkouts/readthedocs.org/user_builds/anyscale-ray/checkouts/35843/python/ray/_private/serialization.py", line 49, in <module>
    from ray.util import inspect_serializability
ImportError: cannot import name 'inspect_serializability' from partially initialized module 'ray.util' (most likely due to a circular import) (/home/docs/checkouts/readthedocs.org/user_builds/anyscale-ray/checkouts/35843/python/ray/util/__init__.py)
```

**Pull request merged**